pkg/controller/operators/olm/apiservices.go

```diff
@@ -35,6 +44,96 @@ func (a *Operator) syncAPIServices(obj interface{}) (syncError error) {
 	return nil
 }
 
+func (a *Operator) shouldRotateCerts(csv *v1alpha1.ClusterServiceVersion) bool {
+	now := metav1.Now()
+	if !csv.Status.CertsRotateAt.IsZero() && csv.Status.CertsRotateAt.Before(&now) {
+		return true
+	}
+
+	return false
+}
+
+// checkAPIServiceResources checks if all expected generated resources for the given APIService exist
+func (a *Operator) checkAPIServiceResources(csv *v1alpha1.ClusterServiceVersion, hashFunc certs.PEMHash) error {
+	for _, desc := range csv.GetOwnedAPIServiceDescriptions() {
+		apiServiceName := fmt.Sprintf("%s.%s", desc.Version, desc.Group)
+		logger := log.WithFields(log.Fields{
+			"csv":        csv.GetName(),
+			"namespace":  csv.GetNamespace(),
+			"apiservice": apiServiceName,
+		})
+
+		// Replace all '.'s with "-"s to convert to a DNS-1035 label
+		serviceName := (strings.Replace(apiServiceName, ".", "-", -1))
+		service, err := a.OpClient.GetService(csv.GetNamespace(), serviceName)
```

Comment:
should we check ownerreferences here?

Reply:
also shouldn't this be querying the lister cache instead of the API?

Reply:
Yes - but I was going to put this off until [ALM-679](https://jira.coreos.com/browse/ALM-679)